src/ape/managers/project/types.py

```diff
@@ -202,6 +202,11 @@ def create_manifest(
                     (self.project_manager.local_project._cache_folder / f"{name}.json").write_text(
                         contract_type.json()
                     )
+                    if self._contracts is None:
```

Comment:
this was a regression from the recent caching improvements we made recently, causing us to sometimes have to force re-compile even when we shouldn't, like when Solidity contracts use libraries